mlflow/sklearn/__init__.py

```diff
@@ -994,3 +986,81 @@ def patched_fit(original, self, *args, **kwargs):
                 safe_patch(
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
+
+
+def eval_and_log_metrics(model, X, y_true, *, prefix, sample_weight=None):
+    """
+    Computes and logs metrics (and artifacts) for the given model and labeled dataset.
+    The metrics/artifacts mirror what is auto-logged when training a model
+    (see mlflow.sklearn.autolog).
+
+    :param model: The model to be evaluated.
+    :param X: The features for the evaluation dataset.
+    :param y_true: The labels for the evaluation dataset.
+    :param prefix: Prefix used to name metrics and artifacts.
+    :param sample_weight: Per-sample weights to apply in the computation of metrics/artifacts.
+    :return: The dict of logged metrics. Artifacts can be retrieved by inspecting the run.
+
+    ** Example **
+
+    .. code-block:: python
+
+        from sklearn.linear_model import LinearRegression
+        import mlflow
+
+        # enable autologging
+        mlflow.sklearn.autolog()
+
+        # prepare training data
+        X = np.array([[1, 1], [1, 2], [2, 2], [2, 3]])
+        y = np.dot(X, np.array([1, 2])) + 3
+
+        # prepare evaluation data
+        X_eval = np.array([[3, 3], [3, 4]])
+        y_eval = np.dot(X_eval, np.array([1,2])) + 3
+
+        # train a model
+        model = LinearRegression()
+        with mlflow.start_run() as run:
+            model.fit(X, y)
+            metrics = mlflow.sklearn.eval_and_log_metrics(model, X_eval, y_eval, prefix="val_")
+
+
+    Each metric's and artifact's name is prefixed with `prefix`, e.g., in the previous example the
+    metrics and artifacts are named 'val_XXXXX'. Note that training-time metrics are auto-logged
+    as 'training_XXXXX'. Metrics and artifacts are logged under the currently active run if one exists,
+    otherwise a new run is started and left active.
+
+    Raises an error if:
+    - prefix is empty
+    - model is not an sklearn estimator or does not support the 'predict' method
```

Comment:
Nit: can we indent the list items here?

```suggestion
    Raises an error if:
      - prefix is empty
      - model is not an sklearn estimator or does not support the 'predict' method
```

![Screen Shot 2021-04-06 at 10 53 14](https://user-images.githubusercontent.com/17039389/113647788-6bba6200-96c6-11eb-8f61-eee414795f4f.png)

https://6278-136202695-gh.circle-artifacts.com/0/docs/build/html/python_api/mlflow.sklearn.html#mlflow.sklearn.eval_and_log_metrics